### PR TITLE
Journal Improvements

### PR DESCRIPTION
**Pull request by calesanz**
Addresses Comments in #10 .
- Add missing route
- Add redirect after entry creation
- Add delete confirmation dialog
- Add textarea for journal content

### CODE REVIEW

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/domain/repository/JournalEntryCrudRepository.java [hunk 1]
```diff
@@ -18,14 +17,8 @@
...
 
-  @Modifying
-  @Transactional
   @PreAuthorize("isAuthenticated()")
-  @Query(value = "insert into journal_entry"
-      + "         (title, content, creation_date, patient_id)"
-      + "         values (:#{#title}, :#{#content}, :#{#creationDate}, (select id from patient where login_username = :#{principal.username}))",
-      nativeQuery = true)
-  void create(String title, String content, Date creationDate);
+  JournalEntry save(JournalEntry entry);
```
Comment (calesanz) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/domain/repository/JournalEntryCrudRepository.java:21]: Gab es einen Grund nicht die Built-In Save Methode zu verwenden?

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/service/JournalEntryService.java [hunk 2]
```diff
@@ -20,17 +22,27 @@ public JournalEntryService(JournalEntryCrudRepository journalEntryCrudRepository
...
-    journalEntryCrudRepository.create(journalEntryDto.getTitle(), journalEntryDto.getContent(), new Date());
+  public Long save(JournalEntryDto journalEntryDto) {
+    Login login = (Login) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    JournalEntry entry = JournalEntry.builder().title(journalEntryDto.getTitle())
+        .content(journalEntryDto.getContent())
+        .creationDate(new Date())
+        .patient(login.getPatient())
+        .build();
+    entry = journalEntryCrudRepository.save(entry);
+    return entry.getId();
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/service/JournalEntryService.java:33]:
```suggestion
    return journalEntryCrudRepository.save(entry)
        .getId();
```

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/service/JournalEntryService.java [hunk 3]
```diff
@@ -20,17 +22,27 @@ public JournalEntryService(JournalEntryCrudRepository journalEntryCrudRepository
...
   }
 
   public void update(JournalEntryDto journalEntryDto) {
-    journalEntryCrudRepository.update(journalEntryDto.getId(), journalEntryDto.getTitle(), journalEntryDto.getContent(), new Date());
+    journalEntryCrudRepository
+        .update(journalEntryDto.getId(), journalEntryDto.getTitle(), journalEntryDto.getContent(),
+            new Date());
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/service/JournalEntryService.java:39]: do not use newlines. @Marcarrian can show you how to update max line length settings.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/service/JournalEntryService.java [hunk 4]
```diff
@@ -20,17 +22,27 @@ public JournalEntryService(JournalEntryCrudRepository journalEntryCrudRepository
...
   }
 
   public JournalEntry findById(Long id) {
     return journalEntryCrudRepository.findById(id)
-        .orElseThrow(() -> new IllegalArgumentException(String.format("JournalEntry with id %s not found", id)));
+        .orElseThrow(() -> new IllegalArgumentException(
+            String.format("JournalEntry with id %s not found", id)));
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/service/JournalEntryService.java:45]: one liner.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java [hunk 5]
```diff
@@ -6,19 +6,24 @@
...
+import com.vaadin.flow.component.textfield.TextArea;
 import com.vaadin.flow.component.textfield.TextField;
 import com.vaadin.flow.data.binder.Binder;
 import com.vaadin.flow.router.BeforeEvent;
 import com.vaadin.flow.router.HasUrlParameter;
 import com.vaadin.flow.router.Route;
 
 @Route("journal")
-public class JournalDetailViewImpl extends SocialAnxietyLayout implements JournalDetailView, HasUrlParameter<Long> {
+public class JournalDetailViewImpl extends SocialAnxietyLayout implements JournalDetailView,
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java:21]: do not break line.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java [hunk 6]
```diff
@@ -44,14 +49,37 @@ public void routeToHomeView() {
...
   }
 
+  @Override
+  public void showDeleteConfirmationDialog() {
+    deleteConfirmationDialog = new Dialog();
+
+    Label messageLabel = new Label();
+    messageLabel.setText("Löschen?");
+
+    Button confirmButton = new Button("Confirm", event ->
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java:59]: maybe `listener::onJournalEntryDeleteConfirm`

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java [hunk 7]
```diff
@@ -44,14 +49,37 @@ public void routeToHomeView() {
...
+        listener.onJournalEntryDeleteConfirm()
+    );
+    Button cancelButton = new Button("Cancel",
+        event -> listener.onJournalEntryDeleteCancel());
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java:63]: maybe `listener::listener.onJournalEntryDeleteCancel`

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java [hunk 8]
```diff
@@ -66,6 +94,7 @@ private FormLayout createFormLayout(Binder<JournalEntryDto> binder) {
     return formlayout;
   }
 
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java:97]: nope.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/domain/repository/JournalEntryCrudRepository.java (hunk 1 above)
Comment (calesanz) [reply source/src/main/java/ch/bfh/bti7081/s2020/blue/domain/repository/JournalEntryCrudRepository.java:22]: Kann save auch ohne PreAuthorize verwendet werden? Wäre dies ein Sicherheitsproblem?

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java (hunk 6 above)
Comment (calesanz) [reply source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java:59]: Does not work unless the Presenter/Listener knows about vaadin.Event stuff.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java (hunk 7 above)
Comment (calesanz) [reply source/src/main/java/ch/bfh/bti7081/s2020/blue/view/journal/JournalDetailViewImpl.java:63]: Does not work unless the Presenter/Listener knows about vaadin.Event stuff.

**Pull request merged**